src/status_im/multiaccounts/key_storage/core.cljs

```diff
@@ -0,0 +1,179 @@
+(ns status-im.multiaccounts.key-storage.core
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.ethereum.mnemonic :as mnemonic]
+            [status-im.multiaccounts.core :as multiaccounts]
+            [status-im.multiaccounts.recover.core :as multiaccounts.recover]
+            [status-im.multiaccounts.model :as multiaccounts.model]
+            [status-im.native-module.core :as native-module]
+            [status-im.navigation :as navigation]
+            [status-im.popover.core :as popover]
+            [status-im.utils.fx :as fx]
+            [status-im.utils.security :as security]
+            [status-im.utils.types :as types]))
+
+(fx/defn key-and-storage-management-pressed
+  "This event can be dispatched before login and from profile and needs to redirect accordingly"
+  {:events [::key-and-storage-management-pressed]}
+  [cofx]
+  (fx/merge cofx
+            (navigation/navigate-to-cofx :key-storage-stack
+                                         {:screen (if (multiaccounts.model/logged-in? cofx)
+                                                    :actions-logged-in
+                                                    :actions-not-logged-in)})))
+
+(fx/defn move-keystore-checked
+  {:events [::move-keystore-checked]}
+  [{:keys [db] :as cofx} checked?]
+  {:db (assoc-in db [:multiaccounts/key-storage :move-keystore-checked?] checked?)})
+
+(fx/defn enter-seed-pressed
+  "User is logged out and probably wants to move multiaccount to Keycard. Navigate to enter seed phrase screen"
+  {:events [::enter-seed-pressed]}
+  [cofx]
+  (fx/merge cofx
+            (navigation/navigate-to-cofx :key-storage-stack
+                                         {:screen :seed-phrase})))
+
+(fx/defn seed-phrase-input-changed
+  {:events [::seed-phrase-input-changed]}
+  [{:keys [db] :as cofx} masked-seed-phrase]
+  (let [seed-phrase (security/safe-unmask-data masked-seed-phrase)]
+    {:db (update db :multiaccounts/key-storage assoc
+                 :seed-phrase (when seed-phrase
+                                (string/lower-case seed-phrase))
+                 :seed-shape-invalid? (or (nil? seed-phrase)
+                                          (empty? seed-phrase)
```

Comment:
You can use `(empty? seed-phrase)` instead of both `(empty? nil) -> true`

Reply:
I added this nil clause while writing tests. Probably mixed some things up, but good catch. This is not needed.